tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/contact/OpenPaasContactsConsumer.java

```diff
@@ -117,16 +130,32 @@ private Mono<EmailAddressContact> messageConsume(AcknowledgableDelivery ackDeliv
             });
     }
 
-    private Mono<EmailAddressContact> handleMessage(ContactAddedRabbitMqMessage contactAddedMessage) {
-        LOGGER.trace("Consumed jCard object message: {}", contactAddedMessage);
-        Optional<ContactFields> maybeContactFields = contactAddedMessage.vcard().asContactFields();
+    private Mono<?> handleMessage(ContactRabbitMqMessage contactMessage, String queue) {
+        LOGGER.trace("Consumed jCard object message: {}", contactMessage);
+        Optional<ContactFields> maybeContactFields = contactMessage.vcard().asContactFields();
+        ContactHandler contactHandler = contactHandlers.get(queue);
+        if (contactHandlers.get(queue) == null) {
+            throw new IllegalArgumentException("Unsupported queue: " + queue);
+        }
+
         return maybeContactFields.map(
-                openPaasContact -> openPaasRestClient.retrieveMailAddress(contactAddedMessage.userId())
+                openPaasContact -> openPaasRestClient.retrieveMailAddress(contactMessage.userId())
                     .map(this::getAccountIdFromMailAddress)
-                    .flatMap(ownerAccountId -> indexContactIfNeeded(ownerAccountId, openPaasContact)))
+                    .flatMap(ownerAccountId -> contactHandler.handle(ownerAccountId, openPaasContact)))
             .orElse(Mono.empty());
     }
 
+    @FunctionalInterface
+    private interface ContactHandler {
+        Mono<?> handle(AccountId ownerAccountId, ContactFields openPaasContact);
+    }
+
+    private final Map<String, ContactHandler> contactHandlers = Map.of(
```

Comment:
OpenPaasContactsConsumer is doing too much and have many reasons to change. We should probably move the logic inside these methods into another class, similar to `com.linagora.tmail.james.jmap.contact.EmailAddressContactMessageHandler`.

Reply:
Unsure of the benefits cost of a split: I like this central place.

IMO let's keep things simple now...